File: src/main/java/com/now/nowbot/model/JsonData/ChuBestPerformance.java
package com.now.nowbot.model.JsonData;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class ChuBestPerformance {
    // 在游戏里的名字
    @JsonProperty("nickname")
    String name;

    // best30 + recent10
    // 这就是 BP
    @JsonProperty("records")
    Records records;

    public record Records(
            @JsonProperty("b30") List<ChuScore> best30,
            @JsonProperty("r10") List<ChuScore> recent10
    ) {}

    // 在查分器里的名字
    @JsonProperty("username")
    String probername;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Records getRecords() {
        return records;
    }

    public void setRecords(Records records) {
        this.records = records;
    }

    public String getProbername() {
        return probername;
    }

    public void setProbername(String probername) {
        this.probername = probername;
    }
}

File: src/main/java/com/now/nowbot/model/JsonData/ChuBestPerformance.kt
package com.now.nowbot.model.jsonData

import com.fasterxml.jackson.annotation.JsonProperty
import com.now.nowbot.model.jsonData.ChuBestPerformance.Records

class ChuBestPerformance {
    // 在游戏里的名字
    @JsonProperty("nickname") var name: String = ""

    // best30 + recent10
    // 这就是 BP
    @JsonProperty("records") var records: Records? = null

    data class Records(
        @JsonProperty("b30") val best30: MutableList<ChuScore> = mutableListOf<ChuScore>(),
        @JsonProperty("r10") val recent10: MutableList<ChuScore> = mutableListOf<ChuScore>(),
    )

    // 在查分器里的名字
    @JsonProperty("username") var probername: String = ""
}
